File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.dart
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   int val;
 *   TreeNode? left;
 *   TreeNode? right;
 *   TreeNode([this.val = 0, this.left, this.right]);
 * }
 */
import 'dart:collection';
class Solution {
  int kthSmallest(TreeNode? root, int k) {
    if(root == null) return -1;
    List<int> list = [];
    Queue<TreeNode> q = Queue();
    q.add(root);
    while(q.isNotEmpty){
        TreeNode cur = q.removeFirst();
        list.add(cur.val);
        if(cur.left != null) q.add(cur.left!);
        if(cur.right != null) q.add(cur.right!);
    }
    list.sort();
    return list[k-1];
  }
}

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> list = new ArrayList<>();
        build(root, list, k);
        return list.get(list.size() -1);
    }

    private void build(TreeNode node, List<Integer> list, int k){
        if(node == null) return;
        build(node.left, list, k);
        if(list.size() < k) {
            list.add(node.val);
        }else {
            return;
        }
        build(node.right, list, k);
    }
}